test/e2e/framework/job/wait.go

```diff
@@ -118,38 +134,52 @@ func WaitForJobSuspend(ctx context.Context, c clientset.Interface, ns, jobName s
 
 // WaitForJobFailed uses c to wait for the Job jobName in namespace ns to fail
 func WaitForJobFailed(c clientset.Interface, ns, jobName string) error {
-	return wait.PollImmediate(framework.Poll, JobTimeout, func() (bool, error) {
-		curr, err := c.BatchV1().Jobs(ns).Get(context.TODO(), jobName, metav1.GetOptions{})
+	return framework.Gomega().Eventually(context.TODO(), framework.HandleRetry(func(ctx context.Context) (*batchv1.Job, error) {
```

Comment:
Try to avoid TODO context. You could extract that change to a separate commit in the PR.

Reply:
Done in a separate commit as suggested 👍🏻